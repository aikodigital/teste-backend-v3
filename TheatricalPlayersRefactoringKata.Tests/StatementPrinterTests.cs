using System;
using System.Collections.Generic;
using ApprovalTests;
using ApprovalTests.Reporters;
using Xunit;

namespace TheatricalPlayersRefactoringKata.Tests;

public class StatementPrinterTests
{
    [Fact]
    [UseReporter(typeof(DiffReporter))]
    public void TestStatementExampleLegacy()
    {
        var plays = new Dictionary<string, Play>();
        plays.Add("hamlet", new Play("Hamlet", 4024, "tragedy"));
        plays.Add("as-like", new Play("As You Like It", 2670, "comedy"));
        plays.Add("othello", new Play("Othello", 3560, "tragedy"));

        Invoice invoice = new Invoice(
            "BigCo",
            new List<Performance>
            {
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40),
            }
        );

        StatementPrinter statementPrinter = new StatementPrinter();
        var result = statementPrinter.Print(invoice, plays);

        Approvals.Verify(result);
    }

    [Fact]
    [UseReporter(typeof(DiffReporter))]
    public void TestTextStatementExample()
    {
        var plays = new Dictionary<string, Play>();
        plays.Add("hamlet", new Play("Hamlet", 4024, "tragedy"));
        plays.Add("as-like", new Play("As You Like It", 2670, "comedy"));
        plays.Add("othello", new Play("Othello", 3560, "tragedy"));
        plays.Add("henry-v", new Play("Henry V", 3227, "history"));
        plays.Add("john", new Play("King John", 2648, "history"));
        plays.Add("richard-iii", new Play("Richard III", 3718, "history"));

        Invoice invoice = new Invoice(
            "BigCo",
            new List<Performance>
            {
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40),
                new Performance("henry-v", 20),
                new Performance("john", 39),
                new Performance("henry-v", 20)
            }
        );

        StatementPrinter statementPrinter = new StatementPrinter();
        var result = statementPrinter.Print(invoice, plays);

        Approvals.Verify(result);
    }
}
