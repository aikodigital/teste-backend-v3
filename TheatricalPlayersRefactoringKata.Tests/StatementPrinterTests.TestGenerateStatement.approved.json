{
    
  "Customer":  "BigCo",
    
  "Items":  [
        
    {
            
      "ItemName":  "Hamlet",
            
      "AmountOwed":  650,
            
      "EarnedCredits":  25,
            
      "Seats":  55
    
        },
        
    {
            
      "ItemName":  "As You Like It",
            
      "AmountOwed":  547,
            
      "EarnedCredits":  12,
            
      "Seats":  35
    
        },
        
    {
            
      "ItemName":  "Othello",
            
      "AmountOwed":  456,
            
      "EarnedCredits":  10,
            
      "Seats":  40
    
        },
        
    {
            
      "ItemName":  "Henry V",
            
      "AmountOwed":  705.4,
            
      "EarnedCredits":  0,
            
      "Seats":  20
    
        },
        
    {
            
      "ItemName":  "King John",
            
      "AmountOwed":  931.6,
            
      "EarnedCredits":  9,
            
      "Seats":  39
    
        },
        
    {
            
      "ItemName":  "Henry V",
            
      "AmountOwed":  705.4,
            
      "EarnedCredits":  0,
            
      "Seats":  20
    
        }
  
    ],
    
  "AmountOwed":  3995.4,
    
  "EarnedCredits":  56

}